public/editor/scripts/editor/js/fc/publisher.js

```diff
@@ -62,29 +62,10 @@ define(function(require) {
       this.updateDialog(publishUrl, true);
     }
 
-    // When there are file events triggered by the editor, update the publish/republish status
-    function handleFileEvent() {
-      if(publisher.needsUpdate) {
-        return;
-      }
-
-      if(!Project.getPublishUrl()) {
-        return;
-      }
-
-      publisher.handlers.unpublishedChangesPrompt();
-      Project.publishNeedsUpdate(true, function(err) {
-        if(err) {
-          console.error("[Thimble] Failed to set the publishNeedsUpdate flag after a file change: ", err);
-          return;
-        }
-        publisher.needsUpdate = true;
-      });
-    }
-    bramble.on("fileChange", handleFileEvent);
-    bramble.on("fileDelete", handleFileEvent);
-    bramble.on("fileRename", handleFileEvent);
-    bramble.on("folderRename", handleFileEvent);
+    bramble.on("fileChange", publisher.showUnpublishedChangesPrompt);
```

Comment:
you'll need to bind this to the publisher instance: `publisher.showUnpublishedChangesPrompt.bind(publisher)`

### Explanation
In javascript, when you pass a function (`showUnpublishedChangesPrompt`) that is bound to an instance (i.e. attached to the prototype of a constructor) (in this case `publisher`) as a parameter when calling another function, it will no longer be bound to that instance (`showUnpublishedChangesPrompt` will not be bound to `publisher`) and so, when the function (`showUnpublishedChangesPrompt`) is called, inside it, `this` will be `undefined`. The javascript `bind` function forces the parameter that you pass in to be bound to a particular instance. You can read a little bit more about it [here](https://developer.mozilla.org/en/docs/Web/JavaScript/Reference/Global_objects/Function/bind)